practice/src/js/main.js

```diff
@@ -14,3 +14,36 @@ productList.renderProducts();
 document
   .querySelector('.nav-list')
   .addEventListener('click', handleNavLinkClick);
+
+/**
+ * Handle click event.
+ * If the user clicks on the delete icon of a product, display a confirmation modal for deletion.
+ * If confirmed, call the API to delete the product. If successful, display a success message and update the product list.
+ * If unsuccessful, display an error message.
+ * @param {Event} event - The click event
+ */
+document.addEventListener('click', async (event) => {
```

Comment:
Should bind the event click after render list product instead of use event listener here